staging/src/k8s.io/apiserver/pkg/endpoints/handlers/response.go

```diff
@@ -479,3 +490,107 @@ func asPartialObjectMetadataList(result runtime.Object, groupVersion schema.Grou
 		return nil, newNotAcceptableError(fmt.Sprintf("no PartialObjectMetadataList exists in group version %s", groupVersion))
 	}
 }
+
+// watchListTransformerFunction an optional function
+// applied to watchlist bookmark events that transforms
+// the embedded object before sending it to a client.
+type watchListTransformerFunction func(watch.Event) watch.Event
+
+// createWatchListTransformerIfRequested returns a transformer function for watchlist bookmark event.
+//
+// note that: for simplicity, nil initialEventsListBlueprint indicates a no-op transformer.
+// otherwise, we assume a watchlist request for which we return a transformer.
+func createWatchListTransformerIfRequested(initialEventsListBlueprint runtime.Object, targetGVK *schema.GroupVersionKind, negotiatedEncoder runtime.Encoder) watchListTransformerFunction {
```

Comment:
Instead of having this nested structure of calls, I suggest removing this one and simply adding this if at the beginning of transform function:
```
if t.initialEventsListBlueprint == nil {
  return event
}
```

Reply:
done